Include/internal/pycore_runtime.h

```diff
@@ -118,25 +120,32 @@ typedef struct pyruntimestate {
 
     struct _Py_unicode_runtime_ids unicode_ids;
 
-    struct _Py_global_objects global_objects;
-    // If anything gets added after global_objects then
-    // _PyRuntimeState_reset() needs to get updated to clear it.
+    struct {
+        // The fields here are values that would otherwise have been
+        // malloc'ed during runtime init in pystate.c and pylifecycle.c.
+        // This allows us to avoid allocation costs during startup and
+        // helps simplify the startup code.
+        struct _Py_global_objects global_objects;
+        struct _is interpreters_main;
+        // The only other values to possibly include here are
+        // mutexes (PyThread_type_lock).  Currently we don't pre-allocate them
+        // because on Windows we only get a pointer type.
+        // All other pointers in PyThreadState are either
+        // populated lazily or change but default to NULL.
+    } _preallocated;
 } _PyRuntimeState;
 
 #define _PyRuntimeState_INIT \
```

Comment:
Might as well remove this as well. It is also only used once.

Reply:
Should `_Py_global_objects_INIT` also move?  It's pretty big and likely to get much bigger.

Reply:
Actually, `_PyRuntimeState_INIT` is actually used twice.

Reply:
FYI, I had been considering moving `_PyRuntimeState_INIT` (and friends) to a separate header file that would only be included by Python/pystate.c and Python/pylifecycle.c.  Doing so would solve several complications.  However, I hesitated because I wasn't sure that was the best approach.